airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -81,44 +82,18 @@ def _get_dep_statuses(
         session: Session,
         dep_context: DepContext,
     ) -> Iterator[TIDepStatus]:
-        # Checking that all upstream dependencies have succeeded
-        if not ti.task.upstream_list:
+        # Checking that all upstream dependencies have succeeded.
+        if not ti.task.upstream_task_ids:
```

Comment:
Minor optimisation: `upstream_list` is a dynamic property, while `upstream_task_ids` is a static set.